### DISCUSSION TITLE
`IBeatmapConverter.CanConvert` does not work as expected

### DISCUSSION THREAD

**Discussion by Flutterish** [ppy/osu#17794] [Q&A] [answered]
When `IBeatmapConverter.CanConvert` returns false, I dont think the beatmap should be shown in the beatmap selection. Instead, it is shown, but runs with the original ruleset for the beatmap. Some of the displayed info, such as star difficulty is still displayed as if it was processed by the ruleset you try to convert to.

The issue seems to be this line https://github.com/ppy/osu/blob/e9018576107564938002ba1413e61a729e7cd152/osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs#L29-L32 which only checks if the user allowed converted beatmaps and not if the beatmap can be converted.

**Comment by frenzibyte** [comment]
`IBeatmapConvert.CanConvert()` currently works in a sense that it would abort conversion gracefully without going through an invalid state, but does not impact the beatmap carousel visually in any way.

**Comment by peppy** [comment]
Can you provide an example of how this manifests visually? I know there are other similar issues which appear when switching rulesets at song select, so would be interested in what particular issue is most important here.

**Comment by Flutterish** [comment]
Yeah, using modified [solosu](https://github.com/Flutterish/Solosu) as an example:
When `CanConvert` always returns true (indicating a ruleset can convert all beatmaps), it looks as you would expect:
![image](https://user-images.githubusercontent.com/40297338/163115440-7147d386-d65a-42ba-99ed-c054145e722d.png)

But when it always returns false (indicating it cant convert any beatmaps "and can only run its native ones") it looks like this:
![image](https://user-images.githubusercontent.com/40297338/163115741-6b36a0a7-896f-4fa8-81e8-d201d2c67ac7.png)
Playing a map in this state launches it in std, which is its native ruleset. I would expect the song select screen to be empty in this case, that is, it should not show the beatmaps that cant be converted.
This of course doesnt apply to just the "always true/false" case, when/if the conversion is lifted from just std, a given ruleset could refuse to convert from some rulesets/formats.

**Comment by peppy** [comment, marked as answer]
Tracking at https://github.com/ppy/osu/issues/17847
